privilege-request-protocol.md

```diff
@@ -0,0 +1,208 @@
+HTTP `Authorization` header using a scheme supported by the server such as
+`Bearer`, or a TLS client certificate.
+
+A client may comprise a software application component that is identified in
+some way to the server. The method by which a software application is identified
+to the server is beyond the scope of this memo. Methods could include, but
+are not limited to, use of the HTTP `Origin` header, or an application identity
+associated in some way with the `Bearer` token.
+
+The method by which a server determines whether an HTTP client has sufficient
+privilege to access the requested resource is beyond the scope of this memo.
+Methods could include, but are not limited to, determining whether the
+authenticated identity is a member of a configured role group, or whether the
+user associated with the authenticated identity has endorsed the identified
+software application currently being used for the access attempted.
+
+The method by which a server determines whether it's appropriate to allow the
+client to request additional privilege is beyond the scope of this memo.
+Methods could include, but are not limited to, determining whether the
+authenticated identity is the controller of the resource, or is the owner of
+the server.
+
+The key words "**MUST**", "**MUST NOT**", "**REQUIRED**", "**SHALL**", "**SHALL
+NOT**", "**SHOULD**", "**SHOULD NOT**", "**RECOMMENDED**", "**NOT RECOMMENDED**",
+"**MAY**", and "**OPTIONAL**" in this document are to be interpreted as
+described in BCP 14 \[[RFC2119][]\] \[[RFC8174][]\] when, and only when, they
+appear in all capitals, as shown here.
+
+Operation
+---------
+The following is a typical exchange according to this protocol:
+
+1. The client attempts access to a resource.
+2. The server determines the client has insufficient privilege and denies the
+   request, returning an HTTP `403 Forbidden` response. The server further
+   determines it's appropriate to allow the client to request additional
+   privilege, so includes in the response a `Link` for the privilege request
+   endpoint URI.
+3. The client determines to request additional privilege and makes a simple
+   `POST` to the provided privilege request endpoint URI without credentials.
+4. The privilege request endpoint determines the request is acceptable and
+   queues the request, returning an HTTP `202 Accepted` response.
+
+The privilege management system might (but is not required to) provide a
+web-based permission management interface, and in some circumstances it might
+be appropriate to allow the client's user to access this interface. The
+privilege request endpoint **MAY** include in its response a URI [template][RFC6570]
+to access this interface, to which the client can redirect the user's trusted
+web browser. The interface URI template **SHOULD** include a `redirect_uri`
+variable so that the interface can eventually redirect back to the client
+application on completion of the management activity.
```

Comment:
hm, but if this is not required, as per line 56, then how would a web app deal with this?

Reply:
if a management URI isn't given, then it's either because a web-based privilege management interface isn't available, or because the server determined it wasn't appropriate to allow the client's user to access the interface. in either case, the web app deals with this by moving on (perhaps by informing the user that a request for access was sent). at this point, until some action out-of-band action is taken, there's nothing else for the app to do with regards to this request.